mantidproject/mantid
# Switch ConfigManager::getValue() to use boost optional return

**Description of work.**
Switch ConfigManager::getValue to use boost optional return instead of using `0/1` return value to indicate success.

This also allows us to use get value or default, tidying up a lot of callers

Additionally a specialisation was added for `bool` which can either accept `true`/`on` or `1` (the former being upper or lower case)

**To test:**

Ensure unit tests pass
Code review

Fixes #22647

Does this update require release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
